lib/carma/models/submission/metadata.rb

```diff
@@ -4,12 +4,42 @@ module CARMA
   module Models
     class Submission
       class Metadata < CARMA::Models::Base
-        request_payload_key :claim_id
+        request_payload_key :claim_id,
+                            :veteran,
+                            :primary_caregiver,
+                            :secondary_caregiver_one,
+                            :secondary_caregiver_two
 
-        attr_reader :claim_id
+        attr_accessor       :claim_id
+
+        attr_reader         :veteran,
+                            :primary_caregiver,
+                            :secondary_caregiver_one,
+                            :secondary_caregiver_two
 
         def initialize(args = {})
           @claim_id = args[:claim_id]
+
+          self.veteran = args[:veteran] || {}
+          self.primary_caregiver = args[:primary_caregiver] || {}
+          self.secondary_caregiver_one = args[:secondary_caregiver_one] if args[:secondary_caregiver_one]
```

Comment:
I think it would be good to add a specific test for one of these cases, because this is a little unclear.
If you are worried that `args[:secondary_caregiver_one]` can be nil, maybe the setter logic should just set it to nil when the value passed to secondary_caregiver_one is nil.  But if `args[:secondary_caregiver_one]` can sometimes be `false`, it would make more sense if I could see a test with the scenario where `false` is this value.

Reply:
The related specs are located here: `spec/lib/carma/models/submission/metadata_spec.rb:51` "::new initializes with defaults".

At the end of the day, I want newly instantiated Metadata models to _always_ to have a Veteran and Caregiver model in the `veteran` and `primaryCaregiver` namespace but default `nil` for the `secondaryCarevier*` namespaces. So I think it makes sense to keep that logic in the initialize method (since it's initialization logic).

I've instead updated the arg check to be more explicit, to show that I'm checking for nil rather than any falsy value. Let me know if that clears up this nil check, or if there is a way to make my intentions more readable.

Reply:
I think making it more explicit is a big improvement.  But I think it makes the setters better/safer if that check is in there.  Currently someone can call `metadata.seconday_caregiver_one = nil` and it will blow up.  So if that logic is in the setter, then you don't need the checks in the initializer which as a bonus knocks out the cognitive complexity check.
I think I'd write the new setter like this, but there's other valid approaches.
```
def secondary_caregiver_one=(sc_one_metadata_hash)
          if sc_one_metadata_hash.nil?
            @secondary_caregiver_one = nil
          else
            @secondary_caregiver_one = Metadata::Caregiver.new(sc_one_metadata_hash)
          end
        end
```

Reply:
That's a good point. I've moved the nil check to the setters and added a spec for setting it to nil.